src/spi.c

```diff
@@ -78,14 +74,14 @@ ErrorStatus spi_set_register(SPI_HandleTypeDef *hspi1, GPIO_TypeDef* CS_GPIO, ui
 }
 
 
-ErrorStatus spi_get_register(SPI_HandleTypeDef *hspi1, GPIO_TypeDef* CS_GPIO, uint16_t CS_GPIO_Pin, uint8_t reg, uint8_t *data, uint8_t bytes_num)
+hub_retcode_t spi_get_register(SPI_HandleTypeDef *hspi1, GPIO_TypeDef* CS_GPIO, uint16_t CS_GPIO_Pin, uint8_t reg, uint8_t *data, uint8_t bytes_num)
 {
-    ErrorStatus ret = SUCCESS;
+    hub_retcode_t ret = OK;
 
-    uint8_t buff = READ_REG(reg);
+    uint8_t buff = SPI_REG_R_M(reg);
     CSB_low(CS_GPIO, CS_GPIO_Pin);
     SPI_send(hspi1, &buff);
-    for(size_t k = 0; k < bytes_num; k++) SPI_recv(hspi1, &data[k]); 
+    for(uint8_t i = 0; i < bytes_num && OK == ret; i++) SPI_recv(hspi1, &data[i]); 
```

Comment:
No i teraz popatrz na tą linię i zastnów się czy jest czytelna xD Tak apropo tych makr

Reply:
ciężko mi wyczuć granice między czytelnością a optymalnością bo wole upchnąć tricky warunek (nie przesadzaj że jest jakiś kosmiczny) niż dokładać linie z osobnym if(sth) return to samo co wyżej
dla mnie np makro
#define CHECK_ERROR_M(error)                    if(OK != (ret |= error)) return ret;
jest mega nieczytelne